gen-compdocs/generators/doc.go

```diff
@@ -36,6 +36,17 @@ func (s byName) Len() int           { return len(s) }
 func (s byName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 func (s byName) Less(i, j int) bool { return s[i].Name() < s[j].Name() }
 
+const generated_warning = `<!--
+The file is auto-generated from the component source code. Please check
+[the overall introduction](/docs/contribute/generate-ref-docs/) for generating
+reference docs and the 
```

Comment:
Could this source share between all reference tools?

Reply:
That might be possible. Can revisit this when revising other tools.